### PR TITLE
ast, plan: return error when the arg of VALUES() function is not a column

### PR DESCRIPTION
**Pull request by XuHuaiyu**
PTAL @eurekaka @winoros 

### What problem does this PR solve? 
``` sql
create table t(a int);
```

Before this PR:
``` sql
select '' as b from t group by values(b);

-- tidb-server will panic
```

After this PR:
```sql
tidb> SELECT '' AS b FROM t GROUP BY VALUES(b);
ERROR 1054 (42S22): Unknown column '' in 'VALUES() function'
```
### What is changed and how it works?
Before this PR, we assume that the argument for VALUES must be a `ColumnNameExpr`,
which is promised as the example above shows.

This PR checks the type of argument for VALUES, if it's not a `ColumnNameExpr`, we will raise an error.

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported function/method change

Side effects

none

Related changes

none

### CODE REVIEW

File: ast/expressions.go [hunk 1]
```diff
@@ -927,7 +927,10 @@ func (n *ValuesExpr) Accept(v Visitor) (Node, bool) {
 	if !ok {
 		return n, false
 	}
-	n.Column = node.(*ColumnNameExpr)
+	n.Column, ok = node.(*ColumnNameExpr)
+	if !ok {
```
Comment (eurekaka) [inline ast/expressions.go]: I prefer to put this check in `ColumnNameExpr` branch of `gbyResolver::Leave`, i.e, for a `ColumnNameExpr`, if the extracted result is not `ColumnNameExpr`, we should report error.

`Accept` acts as a general walker, it should not involve detailed visitor logic IMHO.

File: planner/core/logical_plan_builder.go [hunk 2]
```diff
@@ -1164,6 +1164,10 @@ func (g *gbyResolver) Leave(inNode ast.Node) (ast.Node, bool) {
 			return inNode, false
 		}
 		return ret, true
+	case *ast.ValuesExpr:
+		if v.Column == nil {
+			g.err = ErrUnknownColumn.GenWithStackByArgs("", "VALUES() function")
+		}
```
[the remark this answers is not in the document]
Comment (XuHuaiyu) [reply planner/core/logical_plan_builder.go]: Seems not.
i.e.
MySQL:
``` sql
set @@sql_mode='only_full_group_by';
mysql> SELECT a AS b FROM t GROUP BY VALUES(b);
ERROR 1055 (42000): Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'test.t.a' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
```

If we return ColumnNameExpr here, the check for only_full_group_by will be passed, this is not compatible with MySQL.

**Comment by XuHuaiyu** [comment]
PTAL @eurekaka

File: planner/core/logical_plan_builder.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (XuHuaiyu) [reply planner/core/logical_plan_builder.go]: related issue:
https://github.com/pingcap/tidb/issues/7822

**Pull request merged**